addons/survey/controllers/main.py

```diff
@@ -29,15 +29,24 @@ def _fetch_from_access_token(self, survey_token, answer_token):
         """ Check that given token matches an answer from the given survey_id.
         Returns a sudo-ed browse record of survey in order to avoid access rights
         issues now that access is granted through token. """
-        survey_sudo = request.env['survey.survey'].with_context(active_test=False).sudo().search([('access_token', '=', survey_token)])
+        if not survey_token:
```

Comment:
why not something like that ? (it has the same query count, and we don't bypass the ORM)

```py
def _fetch_from_access_token(self, survey_token, answer_token):
    if not survey_token:
        return request.env['survey.survey'].sudo(), request.env['survey.user_input'].sudo()

    if not answer_token:
        s = request.env['survey.survey'].sudo().search([('access_token', '=', survey_token)], limit=1)
        return s, request.env['survey.user_input'].sudo()

    answer_sudo = request.env['survey.user_input'].sudo().search([
        ('survey_id.access_token', '=', survey_token),
        ('access_token', '=', answer_token)
    ], limit=1)
    survey_sudo = answer_sudo.survey_id
    return survey_sudo, answer_sudo
```

Reply:
Somehow `survey_id.access_token` didn't come to me at that time...
Please do have a look at the rest :)